presto-main/src/main/java/io/prestosql/metadata/MetadataListing.java

```diff
@@ -140,12 +141,21 @@ private MetadataListing() {}
 
         ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> result = ImmutableMap.builder();
         for (Entry<SchemaTableName, List<ColumnMetadata>> entry : tableColumns.entrySet()) {
-            if (allowedTables.contains(entry.getKey())) {
-                result.put(entry.getKey(), accessControl.filterColumns(
-                        session.toSecurityContext(),
-                        new CatalogSchemaTableName(prefix.getCatalogName(), entry.getKey()),
-                        entry.getValue()));
```

Comment:
I think this was motivation why `filterColumns` was accepting `List<ColumnMetadata>` parameter and returning `List<ColumnMetadata>`. It was simply easy to write this particular line and so far this was the only usage of this call.